--- zlatrd.cpp.orig	2012-03-23 10:49:41.000000000 +0100
+++ zlatrd.cpp	2012-03-23 11:07:28.000000000 +0100
@@ -254,7 +254,9 @@
           blasf77_zscal(&i, &tau[i - 1], W(0, iw), &ione);
 
 #if defined(PRECISION_z) || defined(PRECISION_c)
-          blasf77_zdotc(&value, &i, W(0, iw), &ione, A(0, i), &ione);
+          cblas_zdotc_sub(i, W(0, iw), ione,
+			  A(0, i), ione, &value);
+          /* blasf77_zdotc(&value, &i, W(0, iw), &ione, A(0, i), &ione); */
           alpha = tau[i - 1] * -.5f * value;
 #else
           alpha = tau[i - 1] * -.5f * blasf77_zdotc(&i, W(0, iw), &ione, A(0, i), &ione);
@@ -334,11 +336,11 @@
                         out version works with MKL but is not a standard interface
                         for other BLAS zdoc implementations                        
                      */
-                     /*
-                        cblas_zdotc_sub(i_n, W(i +1, i), ione,
-                                        A(i +1, i), ione, &value);
-                     */
-                  blasf77_zdotc(&value, &i_n, W(i+1,i), &ione, A(i+1, i), &ione);
+                     
+	          cblas_zdotc_sub(i_n, W(i +1, i), ione,
+				  A(i +1, i), ione, &value);
+                     
+                  //blasf77_zdotc(&value, &i_n, W(i+1,i), &ione, A(i+1, i), &ione);
                   alpha = tau[i]* -.5f * value;
               #else
                   alpha = tau[i]* -.5f* blasf77_zdotc(&i_n, W(i+1,i), &ione, A(i+1, i), &ione);
